### ISSUE TITLE
Looping file upload

### ISSUE THREAD

**Issue by nanaya** [nextcloud/ios#778]
### Expected behaviour                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
Auto upload one photo.

### Actual behaviour                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Auto upload one photo, repeatedly. Closing the app and starting it again doesn't help. Good thing the instance is on local server.

### Steps to reproduce                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
No idea.

### iOS version
12.1.3

### App version
2.22.8.20


### Server configuration

**Operating system**: FreeBSD

**Web server**: nginx

**Database**: PostgreSQL 11.1

**PHP version**: 7.2.14

**Nextcloud version**: 15.0.2

Activity page showing this:

![](https://s.myconan.net/2019-01/2019-01-24_21.00.50_2805.png)

And mostly that as far as I can scroll down.

**Comment by nanaya** [comment]
Interestingly, browing the Photos directory from inside the app finally stopped the loop.

**Comment by marinofaggiana** [comment]
@nanaya ... strange thing ... I close this and if you want can try the version in TestFlight that has several news for the upload queue, thanks for your report.

**Comment by voggur** [comment]
I have the same issue with the latest version of the app 2.22.8.20.
The server whose on version 13 but I updated it to version 15 in the hope that it would solve the issue.
"browsing Photos directory from inside the app finally stopped the loop." this also works in my case.
I also get allot of "webdav locked: Trying to access locked resource"

**Comment by voggur** [comment]
Can you reopen this issue ?

**Comment by baughj** [comment]
I get same behavior (WebDAV lock / upload loop) with NC 15.0.2 and iOS client 2.22.8.20.

**Comment by djolson11** [comment]
Having the same exact issue. Any news on this?

**Comment by nanaya** [comment]
I think someone needs to try daily build and see if it still happens.

I'll look into installing it sometime later today.

**Comment by voggur** [comment]
Who can I get access to it in Testflight ?

**Comment by baughj** [comment]
This resolved the issue for me. @voggur email ios@nextcloud.com for a TestFlight code.

**Comment by AndrewwHummer** [comment]
Install TestFlight from the Appstore and follow this Link:
https://testflight.apple.com/join/GjNbfo2a

**Comment by marinofaggiana** [comment]
read : https://github.com/nextcloud/ios

**Comment by nariox** [comment]
@marinofaggiana: version 2.22.9 fixes the issue for me. Starting a large camera roll upload still makes the app a little laggy, but the uploads work beautifully.

**Comment by daCaPo** [comment]
Just installed 2.23.0 - the issue still exists :-/. Reopen this issue, or this this a strange race condition between the AppStore version and Github (still showing 2.22.9)?

**Comment by marinofaggiana** [comment]
@daCaPo the version on Master is now 2.23.0, what do you mean for "the issue still exists" please more detail.

**Comment by daCaPo** [comment]
> what do you mean for "the issue still exists" please more detail.

1. I download a file "name.something" and export it to another app for viewing/editing.
1. From the other app, I export the file back to the Nextcloud app (no matter whether via "copy ..." or just "Nextcloud" icon).
1. Nextcloud will upload the file to the folder, but save it as "name-1.something", if "name.something" already exists.

Expected behaviour: Ask the user if the existing file should be overwritten. If no, proceed as above. If yes, just overwrite (and thus keep the version history on the server).

**Comment by nariox** [comment]
Hi @dacapo,
this seems like a completely different issue at all. In fact, it seems like
this is the behavior by design. If you'd like to have an option to
overwrite files instead, you should probably file a new enhancement request
to add this option.

**Comment by daCaPo** [comment]
> this seems like a completely different issue at all

Hmm, I came here via [this post](https://help.nextcloud.com/t/file-upload-problem-with-the-new-ios-app-v2-22-1-0/35474/19) in the forums, claiming this issue would be related to what I described. (The second posting seemingly describes my bug.)

> it seems like this is the behavior by design

In the past, the iOS app just overwrote existing files without any question to the user, so it used to work. Only the "ask the user" part of the [expected behaviour](https://github.com/nextcloud/ios/issues/778#issuecomment-465209005) would be a feature request.
